### PR TITLE
perf: cli snapshot warmup

### PR DESCRIPTION
**Pull request by littledivy**
Depends on https://github.com/denoland/deno_core/pull/570

### CODE REVIEW

**Comment by littledivy** [comment]
Seeing a consistent 1-2ms improvement on Linux

```
divy@divy:~/gh/perf-toolkit$ ./startup-compare-deno.sh
Benchmark 1 (190 runs): deno run test.js
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          26.2ms ± 2.25ms    24.4ms … 42.7ms         12 ( 6%)        0%
  peak_rss           51.1MB ±  192KB    50.6MB … 52.4MB          2 ( 1%)        0%
  cpu_cycles         55.4M  ± 4.79M     51.8M  … 94.2M          11 ( 6%)        0%
  instructions        109M  ± 14.2K      109M  …  110M           6 ( 3%)        0%
  cache_references   2.52M  ± 46.6K     2.43M  … 2.75M           8 ( 4%)        0%
  cache_misses        545K  ± 13.4K      497K  …  639K          12 ( 6%)        0%
  branch_misses       334K  ± 4.04K      327K  …  357K           4 ( 2%)        0%
Benchmark 2 (206 runs): /home/divy/gh/littledivy/deno/target/release/deno run test.js
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          24.3ms ± 1.08ms    23.2ms … 31.0ms         17 ( 8%)        ⚡-  7.2% ±  1.3%
  peak_rss           52.8MB ±  235KB    52.3MB … 53.8MB          2 ( 1%)        💩+  3.3% ±  0.1%
  cpu_cycles         50.7M  ± 2.52M     48.6M  … 64.8M          21 (10%)        ⚡-  8.5% ±  1.3%
  instructions        103M  ± 12.0K      103M  …  103M           2 ( 1%)        ⚡-  5.8% ±  0.0%
  cache_references   2.43M  ± 42.5K     2.33M  … 2.78M           5 ( 2%)        ⚡-  3.5% ±  0.3%
  cache_misses        510K  ± 11.8K      472K  …  605K          13 ( 6%)        ⚡-  6.5% ±  0.5%
  branch_misses       310K  ± 4.56K      304K  …  341K          10 ( 5%)        ⚡-  7.2% ±  0.3%
```

File: runtime/js/99_main.js [hunk 1]
```diff
@@ -657,118 +657,124 @@ function bootstrapMainRuntime(runtimeOptions) {
...
-    name: core.propWritable(""),
-    close: core.propWritable(windowClose),
-    closed: core.propGetterOnly(() => windowIsClosing),
-  });
-  ObjectSetPrototypeOf(globalThis, Window.prototype);
+  // We don't want to set up the global object if we're just warming up
+  // the bootstrap process.
+  //
+  // TODO(@littledivy): Investigate more into warming up more parts.
+  if (!warmup) {
```
Comment (bartlomieju) [inline runtime/js/99_main.js:664]: Maybe invert it and exit early?

**Review by bartlomieju — COMMENTED** [review]
The failures in `make_context` look suspicious. Could you remove the unwrap there and surface errors?

File: runtime/js/99_main.js (hunk 1 above)
Comment (littledivy) [reply runtime/js/99_main.js:664]: For some reason, inverting the control flow does not cause the code inside the `if` to compile.

**Comment by littledivy** [comment]
The node:vm errors are because the runtime_cb is not executed in deno_core during snapshot warmup. I'll fix it up in deno_core.

File: runtime/js/99_main.js (hunk 1 above)
Comment (bartlomieju) [reply runtime/js/99_main.js:664]: I see, maybe leave a comment about it?

**Comment by bartlomieju** [comment]
Should we close this one, now that #22714 has landed?

**Comment by littledivy** [comment]
Yup

**Pull request closed without merging**